src/Components/MainTableView/UpdateTuple/UpdateTuple.tsx

```diff
@@ -71,13 +71,73 @@ class UpdateTuple extends React.Component<{token: string, selectedSchemaName:str
+                  if (attributeKeyVal[0] === attr.attributeName) { // attributeKeyVal[0] is the key
+                    let dateFormat = new Date(attributeKeyVal[1])
+                    tupleBuffer[attributeKeyVal[0]] = dateFormat.toISOString().split('T')[0];
+                  }
+                })
+              }
+              else if (attr.constructor === SecondaryTableAttribute) {
+                Object.entries(columns.secondaryEntries).forEach((attributeKeyVal: any) => {
+                  if (attributeKeyVal[0] === attr.attributeName) {
+                    let dateFormat = new Date(attributeKeyVal[1])
+                    tupleBuffer[attributeKeyVal[0]] = dateFormat.toISOString().split('T')[0];
+                  }
+                })
+              }
+            }
+            else if (attr.attributeType === TableAttributeType.DATETIME || attr.attributeType === TableAttributeType.TIMESTAMP) {
+              if (attr.constructor === PrimaryTableAttribute) {
+                Object.entries(columns.primaryEntries).forEach((attributeKeyVal: any) => {
+                  if (attributeKeyVal[0] === attr.attributeName) {
+                    let dateFormat = new Date(attributeKeyVal[1])
+                    tupleBuffer[attributeKeyVal[0]] = dateFormat.toISOString().split('T').join(' ').split('.')[0];
+                  }
+                })
+              }
+              else if (attr.constructor === SecondaryTableAttribute) {
+                Object.entries(columns.secondaryEntries).forEach((attributeKeyVal: any) => {
+                  if (attributeKeyVal[0] === attr.attributeName) {
+                    let dateFormat = new Date(attributeKeyVal[1])
+                    tupleBuffer[attributeKeyVal[0]] = dateFormat.toISOString().split('T').join(' ').split('.')[0];
+                  }
+                })
+              }
+            }
+            else if (attr.attributeType === TableAttributeType.TIME) {
+              if (attr.constructor === PrimaryTableAttribute) {
+                Object.entries(columns.primaryEntries).forEach((attributeKeyVal: any) => {
+                  if (attributeKeyVal[0] === attr.attributeName) {
+                    tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+                  }
+                })
+              }
+              else if (attr.constructor === SecondaryTableAttribute) {
+                Object.entries(columns.secondaryEntries).forEach((attributeKeyVal: any) => {
+                  if (attributeKeyVal[0] === attr.attributeName) {
+                    tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+                  }
+                })
+              }
+            }
+          })
+        });
+        Object.entries(columns.primaryEntries).forEach((attributeKeyVal: Array<any>) => {
+          if (!tupleBuffer[attributeKeyVal[0]]) {
+            tupleBuffer[attributeKeyVal[0]] = attributeKeyVal[1]
+          }
```

Comment:
Okay yeah I am not letting this fly this time, this code seems to be a copy of code in the insert function.

Reply:
we worked hard on this.